v2/introspector.go

```diff
@@ -0,0 +1,68 @@
+package car
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/ipfs/go-cid"
+	carv1 "github.com/ipld/go-car"
+	internalio "github.com/ipld/go-car/v2/internal/io"
+	"io"
+)
+
+// Introspection captures the result of an Introspect call.
+type Introspection struct {
```

Comment:
isn't this very similar to the header type?

Reply:
That's the thing; which header type? If `CarHeader` in v1 then that doesn't have offsets, if `Header` in v2 then that doesn't have roots nor the version. Not sure what changes would come in v3 but the chances are it would have stuff that are only special to v3. How would we unify them all I asked myself?

I thought a dedicated `Introspection` type could tell us things we want to know about _a_ CAR regardless of its version.

Reply:
Ah, I see. Maybe embed v2's header struct type, and add the extra fields to the parent like version and roots? That way, at least, you don't duplicate the header fields.

I don't think you need to worry about forwards compatibility with v3. If/when that comes, I seem to understand that many pieces of the core design would change, so it's likely that we would need a carv3 module anyway.

Reply:
OK that makes sense. I embedded `carv1.CarHeader` into `Introspect` too while at it.

Reply:
Beware that embedding multiple structs can get ugly, e.g. if any names conflict they will not be promoted.

Reply:
Do you think we should only embed car v2 header?

Reply:
This is a carv2 module, so my intuition would be that APIs should generally be geared towards the v2 header, since the v2 spec is generally a superset of the v1 spec :) I ultimately defer to you on this though, as I haven't thought hard about this API. Maybe a flat struct without embedding is best in your opinion - I just thought that duplicating many of the fields looked a bit off.